idaes/surrogate/keras_surrogate.py

```diff
@@ -0,0 +1,201 @@
+        net = load_keras_sequential(self._keras_model, omlt_scaling, self.input_bounds())
+
+        if formulation == 'full-space':
+            formulation_object = FullSpaceContinuousFormulation(net)
+        elif formulation == 'reduced-space':
+            formulation_object = ReducedSpaceContinuousFormulation(net)
+        elif formulation == 'relu-bigm':
+            formulation_object = ReLUBigMFormulation(net)
+        elif formulation == 'relu-complementarity':
+            formulation_object = ReLUComplementarityFormulation(net)
+
+        block.neural_network.build_formulation(formulation_object,
+                                               input_vars=block._input_vars_as_list(),
+                                               output_vars=block._output_vars_as_list())
+
+
+
+    def evaluate_surrogate(self, inputs):
+        """
+        Method to evaluate Keras model at a set of input values.
+
+        Args:
+            inputs: numpy array of input values. First dimension of array
+                must match the number of input variables.
+
+        Returns:
+            outputs: numpy array of values for all outputs evaluated at input
+                points.
+        """
+        x = inputs
+        if self._input_scaler is not None:
+            x = self._input_scaler.scale(x)
+            
+        y = self._keras_model.predict(x.to_numpy())
+
+        # y is a numpy array, make it a dataframe
+        y = pd.Dataframe(data=y, columns=self.output_labels(), index=inputs.index)
+
+        if self._output_scaler is not None:
+            y = self._output_scaler.unscale(y)
+
+        return y
+
+    @classmethod
+    def save_to_file(cls, keras_folder_name):
+        """
+        Save the surrogate object to disk by providing the name of the
+        folder to contain the keras model and additional IDAES metadata
+
+        Args:
+           folder_name: str
+              The name of the folder to contain the Keras model and additional
+              IDAES metadata
+        """
+        self._keras_model.save(keras_folder_name)
+        info = dict()
+        info['input_scaler'] = None
+        if self._input_scaler is not None:
+            info['input_scaler'] = self._input_scaler.to_dict()
+        info['output_scaler'] = None
+        if self._output_scaler is not None:
+            info['output_scaler'] = self._output_scaler.to_dict()
+        
+        # serialize information from the base class
+        info['input_labels'] = self.input_labels()
+        info['output_labels'] = self.output_labels()
+        info['input_bounds'] = self.input_bounds()
+
+        with open(os.path.join(keras_folder_name, 'idaes_info.json'), 'w') as fd:
+            json.dump(info, fd)
+
+    @classmethod
+    def load_from_file(cls, keras_folder_name):
+        """
+        Load the surrogate object from disk by providing the name of the
+        folder holding the keras model
+
+        Args:
+           folder_name: str

```

Comment:
Similar to my comment above, I think this should take an archive file (ZIP?) as the argument and not a directory.

Reply:
I agree we should look at how we serialize all of the surrogate objects in a future PR.